cmd/state-svc/test/integration/svc_int_test.go

```diff
@@ -0,0 +1,114 @@
+package integration
+
+import (
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/exeutils"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/svcctl"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/shirou/gopsutil/process"
+	"github.com/stretchr/testify/suite"
+)
+
+type SvcIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *SvcIntegrationTestSuite) TestStartStop() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("start"))
+	cp.Expect("Starting")
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Checking")
+	cp.ExpectRe("Port:\\s+:\\d+")
+	cp.ExpectRe("Log:\\s+.+?\\.log")
+	cp.ExpectExitCode(0)
+
+	sockFile := svcctl.NewIPCSockPathFromGlobals().String()
+	suite.True(fileutils.TargetExists(sockFile))
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("stop"))
+	cp.Expect("Stopping")
+	cp.ExpectExitCode(0)
+
+	suite.False(fileutils.TargetExists(sockFile))
+}
+
+func (suite *SvcIntegrationTestSuite) TestSignals() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	// SIGINT (^C)
+	cp := ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("foreground"))
+	cp.Expect("Starting")
+	cp.Signal(syscall.SIGINT)
+	cp.Expect("caught a signal: interrupt")
+	cp.ExpectNotExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
```

Comment:
We should also check that the socket file has been cleaned up for the SIGINT case.